Code/GraphMol/FMCS/FMCS.cpp

```diff
@@ -353,6 +369,140 @@ bool MCSBondCompareOrderExact(const MCSBondCompareParameters& p,
   return false;
 }
 
+//=== RING COMPARE ========================================================
+inline static bool ringFusionCheck(const short unsigned c1[],
+    const short unsigned c2[], const ROMol& mol1,
+    const FMCS::Graph& query, const ROMol& mol2,
+    const FMCS::Graph& target, const MCSParameters* p) {
+  const RingInfo *ri2 = mol2.getRingInfo();
+  const VECT_INT_VECT &br2 = ri2->bondRings();
+  std::vector<size_t> nonFusedBonds(br2.size(), 0);
+  std::vector<size_t> numMcsBondRings(br2.size(), 0);
+  RDKit::FMCS::Graph::BOND_ITER_PAIR bpIter = boost::edges(query);
+  size_t i = 0;
+  // numMcsBondRings stores the number of bonds which
+  // are part of the MCS for each ring
+  for (auto it = bpIter.first; it != bpIter.second; ++it) {
+    const Bond *b = mol2.getBondBetweenAtoms(
+      target[c2[boost::source(*it, query)]], target[c2[boost::target(*it, query)]]);
+    if (!b) continue;
+    unsigned int bi = b->getIdx();
+    if (!ri2->numBondRings(bi)) continue;
+    for (i = 0; i < br2.size(); ++i) {
+      if (std::find(br2[i].begin(), br2[i].end(), bi) != br2[i].end())
+        ++numMcsBondRings[i];
+    }
+  }
+  // nonFusedBonds stores the number of non-fused bonds
+  // (i.e., which belong to a single ring) for each ring
+  for (i = 0; i < br2.size(); ++i) {
+    for (auto bi: br2[i]) {
+      if (ri2->numBondRings(bi) == 1)
+        ++nonFusedBonds[i];
+    }
+  }
+  /* if a ring has at least one bond which is part of the MCS,
+     we need to check how many bonds are actually part of the MCS:
+     if they are greater than the number of fused bonds but lower
+     than the number of non-fused bonds, then the ring is not
+     complete and we should return true straightaway
+     We need to check that the number of bonds in a ring
+     is greater than the number of fused bonds because that
+     guarantees that this ring is actually involved in the
+     MCS and not just adjacent to another ring which is
+     indeed part of the MCS. */
+  bool missingFusedBond = false;
+  for (i = 0; i < br2.size(); ++i) {
+    if (numMcsBondRings[i]
+      && numMcsBondRings[i] > (br2[i].size() - nonFusedBonds[i])
+      && numMcsBondRings[i] < nonFusedBonds[i])
+      break;
+    if (numMcsBondRings[i] == nonFusedBonds[i]
+      && numMcsBondRings[i] < br2[i].size())
+      missingFusedBond = true;
+  }
+  if (i < br2.size())
+    return true;
+  /* If we found that the MCS is missing a fused bond, we may need to
+     check against the smaller molecule as well.
+     Consider this case: MCS between
+     2-methylbicyclo[4.3.0]nonane and 1-methylbicyclo[3.1.0]hexane
+
+                   \__
```

Comment:
nice ascii art!
:smile:

Reply:
The cyclopropane could be better, but... :-)